src/sentry/snuba/entity_subscription.py

```diff
@@ -69,6 +69,11 @@ def apply_dataset_query_conditions(
     return event_type_conditions
 
 
+class _EntitySpecificParams(TypedDict):
```

Comment:
@wedamija let me know if you think this is a good alternative

Reply:
That seems like a reasonable compromise. I feel like I remember people saying not to use `TypedDict` for some reason, but I don't remember what it was.